File: genprime.php
#!/usr/bin/php
<?php

function isprime($x)
{
	if ($x < 2)
		return false;
	if ($x == 2)
		return true;
	if ($x % 2 == 0)
		return false;
	if ($x < 9)
		return true;
	if (($x + 1) % 6 != 0)
		if (($x - 1) % 6 != 0)
			return false;
	$lim = sqrt($x) + 1.0;
	for ($y = 3; $y < $lim; $y += 2)
	{
		if ($x % $y == 0)
			return false;
	}
	return true;
}

function genprime($max)
{
	$count = 0;
	$current = 1;
	while ($count < $max)
	{
		if (isprime($current))
			$count++;
		$current++;
	}
	return $current - 1;
}

$start = $argc > 1 ? (int)($argv[1]) : 0;
$stop = $argc > 2 ? (int)($argv[2]) + 1 : 0;
for ($x = $start; $x < $stop; $x += $start)
{
	$begin = microtime(true);
	$last = genprime($x);
	$end = microtime(true);
	$duration = $end - $begin;
	printf ("Found %8d primes in %10.5f seconds (last was %10d)\n",
		$x, $duration, $last);
}

?>

File: genprime.pl
#!/usr/bin/perl

use strict;
use warnings;

use Time::HiRes;

sub isprime {
	my ($x, $lim, $y);
	$x = $_[0];
	if ($x < 2) {
		return 0;
	}
	if ($x == 2) {
		return 1;
	}
	if ($x % 2 == 0) {
		return 0;
	}
	if ($x < 9) {
		return 1;
	}
	if (($x + 1) % 6 != 0) {
		if (($x - 1) % 6 != 0) {
			return 0;
		}
	}
	$lim = sqrt($x) + 1.0;
	for ($y = 3; $y < $lim; $y += 2)
	{
		if ($x % $y == 0)
		{
			return 0;
		}
	}
	return 1;
}

sub genprime {
	my ($max, $count, $current);
	$max = $_[0];
	$count = 0;
	$current = 1;
	while ($count < $max)
	{
		if (isprime($current) != 0)
		{
			$count = $count + 1;
		}
		$current = $current + 1;
	}
	return $current - 1;
}

my $argc = @ARGV + 1;

my ($start, $stop);

if ($argc > 1) {
	$start = $ARGV[0] + 0;
} else {
	$start = 0;
}
if ($argc > 2) {
	$stop = $ARGV[1] + 1;
} else {
	$stop = 0;
}

my ($x, $begin, $end, $duration, $last);

for ($x = $start; $x < $stop; $x = $x + $start)
{
	$begin = Time::HiRes::time();
	$last = genprime($x);
	$end = Time::HiRes::time();
	$duration = $end - $begin;
	printf "Found %8d primes in %10.5f seconds (last was %10d)\n",
		$x, $duration, $last;
}
